docker/docker_client.go

```diff
@@ -243,7 +243,13 @@ func (c *dockerClient) getBearerToken(realm, service, scope string) (string, err
 	return tokenStruct.Token, nil
 }
 
-func getAuth(hostname string) (string, string, error) {
+func getAuth(ctx *types.SystemContext, hostname string) (string, string, error) {
+	if ctx.DockerAuthConfig != nil {
+		if ctx.DockerAuthConfig.Auth != "" {
+			return decodeDockerAuth(ctx.DockerAuthConfig.Auth)
+		}
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
```

Comment:
> K8s and OpenShift can just provide Auth. 

AFAICT for access to the Docker endpoint for OpenShift, one would set `{Username:"unused",Password:openShiftTokenString}`, at least based on what `~/.kube/config` is storing. Does the API really give the user the Docker base64 blob somewhere?